jose/constants.py

```diff
@@ -12,12 +12,15 @@ class Algorithms(object):
     ES256 = 'ES256'
     ES384 = 'ES384'
     ES512 = 'ES512'
+    # RFC8037 - https://tools.ietf.org/html/rfc8037
+    EdDSA = 'EdDSA'
 
     HMAC = {HS256, HS384, HS512}
     RSA = {RS256, RS384, RS512}
     EC = {ES256, ES384, ES512}
+    ED = {EdDSA}
 
-    SUPPORTED = HMAC.union(RSA).union(EC)
+    SUPPORTED = HMAC.union(RSA).union(EC).union(ED)
```

Comment:
I cannot import anything from `jose.backends` here, because all backends import from `jose.constants`, so `from jose.backends import Ed25519Key` here creates a circular import.

To work around this, the `nacl_backend` modifies the `ALGORITHMS.SUPPORTED` and `ALGORITHMS.ALL` variables directly. This is not ideal, but it's the only way (that I know of) to implement this in a decoupled manner, and the best way that supports Python < 3.6.

On that note, the [`__init_subclass__` hook](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) is probably the best way to implement this in Python 3.6+.